addons/promote/common/src/main/java/org/commonjava/indy/promote/validate/PromotionValidationTools.java

```diff
@@ -595,20 +602,34 @@ public Transfer getTransfer( final StoreResource resource )
 
     private <T> void runParallelInBatchAndWait( Collection<Collection<T>> batches, Closure closure, Logger logger )
     {
-        final CountDownLatch latch = new CountDownLatch( batches.size() );
-        batches.forEach( batch -> ruleParallelExecutor.execute( () -> {
-            try
-            {
-                logger.trace( "The paralleled exe on batch {}", batch );
-                batch.forEach( e -> closure.call( e ) );
-            }
-            finally
-            {
-                latch.countDown();
-            }
-        } ) );
+        ThreadContext ctx = ThreadContext.getContext( true );
+        AtomicInteger depth = (AtomicInteger) ctx.computeIfAbsent( ITERATION_DEPTH, k -> new AtomicInteger( -1 ) );
+        depth.incrementAndGet();
```

Comment:
Little confused here. Seems this depth is not used in this code block?

Reply:
And if I understand correctly, it is always 0 (only increment once?). So what is it used for?

Reply:
This is a coding mistake, I forgot to use it below. It will always be zero as long as we don't have nested iteration. If we do, it'll show up. I think we could use this to help us find problems like the nested parallelEach problem we found recently...